float32 x_
float32 y_
float32[] type_probability_vector_